### DISCUSSION TITLE
Support for Bambu Lab printers

### DISCUSSION THREAD

**Discussion by PPCM** [fdm-monster/fdm-monster#3581] [General]
Hello the team,

Do you think it will be possible to support Bambu Lab printers?
In my case, I have some Octoprint printers, now I bougth some X1C and A1 printers, and it will be great if they could be added to my instance of FDM Monster

I saw an Octoprint plugin (https://github.com/jneilliii/OctoPrint-BambuPrinter) to connect BambuLab printers to OctoPrint, but it implies an Octoprint by printer, so it will become difficult to manage - I didn't try it

What do you think about that?

Thanks for your work

**Comment by davidzwa** [comment]
Hi there,

I've recently done a study to looking at supporting bambu labs printers. Its quite a bit of work as Bambu requires MQTT and (s)FTP connections. This is definitely possible, but just a lot of work to develop, test and polish.

(Also consider that A1 runs over WiFi only, so do take a serious look at your infrastructure limitations!)

For now I'd be open for contributions (by other developers out there) to take up this task. I'll be available on the Discord server or here on Github to provide guidance as to how this is best done.
